feature/text_based.py

```diff
@@ -0,0 +1,703 @@
+        self.matrix = None
+        self.features = None
+        self._num_rows = None
+        self._num_cols = None
+
+    def run_content_selection(self, input_df: pd.DataFrame, categories: pd.DataFrame, selection_size: int,
+                              featurization_method: TextWiser, optimization_method: str = "exact",
+                              cost_metric: str = "diverse", trials: int = 10) -> List:
+
+        """Run content selection algorithm.
+
+        Parameters
+        ----------
+        input_df: pd.DataFrame
+            Input data frame with categories and features of content to select from.
+
+        categories: pd.DataFrame
+            Data frame with data that contains categories/labels to be covered.
+
+        selection_size: int
+            Number of feature to select.
+
+        featurization_method: TextWiser
+
+        optimization_method: str, default="exact"
+            Optimization method used to perform content selection.
+                - See TextBased in Selector.py for supported options definition
+
+        cost_metric: str, default = "diverse"
+
+        trials: int, default = 10
+
+        Returns
+        -------
+        List of indices in data that are selected.
+        """
+
+        # Initialize arguments
+        self.selection_size = selection_size
+        self.featurization_method = featurization_method
+        self.cost_metric = cost_metric
+
+        # Process data
+        self._process_data(input_df, categories, selection_size, optimization_method, cost_metric)
+
+        if optimization_method == "exact":
+            selected = self._select_multi_level_optimization()
+        elif optimization_method == "greedy":
+            selected = self._select_greedy()
+        elif optimization_method == "kmeans":
+            selected = self._select_kmeans()
+        elif optimization_method == "random":
+            selected = self._select_random(trials=trials)
+        else:
+            raise NotImplementedError(f"{optimization_method} has not been implemented.")
+
+        # Sort selection
+        selected.sort()
+
+        return selected
+
+    def _process_data(self, input_df: pd.DataFrame, categories: pd.DataFrame, selection_size: int,
+                      optimization_method: str, cost_metric: str) -> NoReturn:
+        """_process_data
+        We have implemented two versions of this function. In the version here, the categories parameter
+        actually refers to the labels (rather than categories) to be covered. It generates features for input data
+        using a specified featurization method, and creates a matrix where each row corresponds to a label and each
+        column corresponds to a feature.
+        The version of code for generating labels from categories has been moved to utils.py.
+
+        Parameters
+        ----------
+        input_df: pd.DataFrame
+            Input data frame with categories and features of content to select from.
```

Comment:
I don't think there is any categories involved here?

Reply:
Fixed